mailbox/hbase/src/main/java/org/apache/james/mailbox/hbase/mail/HBaseMessageMapper.java

```diff
@@ -189,12 +193,12 @@ public void endRequest() {
         return messageList;
     }
 
-    private List<MailboxMessage> findMessagesInMailboxAfterUID(HBaseId mailboxId, long from, int batchSize, boolean flaggedForDelete) throws IOException {
+    private List<MailboxMessage> findMessagesInMailboxAfterUID(HBaseId mailboxId, MessageUid messageUid, int batchSize, boolean flaggedForDelete) throws IOException {
         List<MailboxMessage> messageList = new ArrayList<MailboxMessage>();
         HTable messages = new HTable(conf, MESSAGES_TABLE);
         // uids are stored in reverse so we need to search
-        Scan scan = new Scan(messageRowKey(mailboxId, Long.MAX_VALUE),
-                messageRowKey(mailboxId, from - 1));
+        Scan scan = new Scan(messageRowKey(mailboxId, MessageUid.MAX_VALUE),
+                messageRowKey(mailboxId, messageUid.previous()));
```

Comment:
should break when messageUid is 1, because it should call minMessageRowKey instead